cmd/edit/machinepool/cmd.go

```diff
@@ -180,12 +181,8 @@ func run(cmd *cobra.Command, argv []string) {
 		}
 	}
 
-	var err error
-	if cluster.Hypershift().Enabled() {
-		err = editNodePool(cmd, machinePoolID, clusterKey, cluster, r)
-	} else {
-		err = editMachinePool(cmd, machinePoolID, clusterKey, cluster, r)
-	}
+	service := machinepool.NewMachinePoolService()
```

Comment:
@hunterkepley Not for this PR, but I've been thinking it would be good for us to have a pattern to be able to mock out the services that are created like this with an uber mock. It would make testing the `runner` much easier. My initial thought is to put the service onto the cmd options that get passed in so for testing we can explicitly set the mock, and for normal execution, we just set the correct implementation or have a pattern whereby if the service is nil, we create one.